RELEASE/scripts/autoscend/auto_quest_level_12.ash

```diff
@@ -1290,6 +1299,223 @@ boolean L12_themtharHills()
 	return true;
 }
 
+boolean L12_farm()
+{
+	if(get_property("auto_skipL12Farm").to_boolean())
+	{
+		return false;
+	}
+	if(get_property("sidequestFarmCompleted") != "none")
+	{
+		set_property("auto_skipL12Farm", "true");
+		return false;
+	}
+	if(internalQuestStatus("questL12War") != 1)
+	{
+		return false;
+	}
+	
+	string faction = "fratboy";
+	if(get_property("auto_hippyInstead").to_boolean())
+	{
+		faction = "hippy";
+	}
+	
+	if(faction == "fratboy" && get_property("hippiesDefeated").to_int() < 458)
+	{
+		return false;
+	}
+	
+	//calculate if farm is worth doing.
+	int adv_saved = 0;
+	int sidequests_done = 0;
+	int enemiesRemaining = 1000;
+	int adv_needed = 40;
+	
+	//calculate adventures needed
+	if(get_property("chaosButterflyThrown").to_boolean())
+	{
+		adv_needed -= 15;
+	}
+	//TODO does this count account for free fights in those zones? needs testing
+	adv_needed -= $location[McMillicancuddy's Barn].turns_spent;
+	adv_needed -= $location[McMillicancuddy's Pond].turns_spent;
+	adv_needed -= $location[McMillicancuddy's Back 40].turns_spent;
+	adv_needed -= $location[McMillicancuddy's Other Back 40].turns_spent;
+	
+	//calculated sidequests
+	if(get_property("sidequestArenaCompleted") == faction)
+	{
+		sidequests_done++;
+	}
+	if(get_property("sidequestJunkyardCompleted") == faction)
+	{
+		sidequests_done++;
+	}
+	if(get_property("sidequestLighthouseCompleted") == faction)
+	{
+		sidequests_done++;
+	}
+	if(get_property("sidequestOrchardCompleted") == faction)
+	{
+		sidequests_done++;
+	}
+	if(get_property("sidequestNunsCompleted") == faction)
+	{
+		sidequests_done++;
+	}
+	
+	int base_kills = 2^sidequests_done;
```

Comment:
Use `**` not `^`, because `^` is bitwise XOR: https://wiki.kolmafia.us/index.php?title=Operators

Reply:
commit 7dedcbb